cityofaustin/vision-zero
# Ability to global search by case ID

## Associated issues
- https://github.com/cityofaustin/atd-data-tech/issues/19478

I'm curious how folks feel about this. It can be a little clunky, but was the easiest path forward I could come up with. 

If you look in the commit history you will see that I played around with using [react-select](https://react-select.com/) to show search results in a dropdown menu, but i scrapped this because it was getting pretty complicated.

<img width="324" alt="Screenshot 2024-10-15 at 3 48 29 PM" src="https://github.com/user-attachments/assets/fa2bff57-08a0-4c99-8ee6-8766a8c003e9">


## Testing

**URL to test:** 
[Deploy preview](https://deploy-preview-1579--atd-vze-staging.netlify.app/#/crashes)

**Steps to test:**
1. Search with a valid crash ID
2. Search with an invalid crash ID, then edit the crash ID to be valid
3. Repeat with case ID
4. Try searching by **Crash ID** with a valid **Case ID**, then use the button to switch the search field to **Case ID**
5. Set the search field to **Case ID** and refresh the page to verify that your field selection is persisted

---

#### Ship list
